boards/arm/thingy91_nrf9160/adp5360_init.c

```diff
@@ -13,6 +13,13 @@ LOG_MODULE_REGISTER(board_secure, CONFIG_BOARD_LOG_LEVEL);
 #define ADP536X_I2C_DEV_NAME	DT_LABEL(DT_NODELABEL(i2c2))
 #define LC_MAX_READ_LENGTH	128
 
+/* The BH1749 on must be powered by the ADP5360 before it can be initialized. */
+#if IS_ENABLED(CONFIG_BH1749)
+BUILD_ASSERT(CONFIG_BOARD_INIT_PRIORITY < CONFIG_SENSOR_INIT_PRIORITY,
+	     "The board initialization must happen before the sensor initialization "
+	     "for the sensors to be powered before initializing");
```

Comment:
```suggestion
	     "The configured init priorities do not allow the board initialization before sensor initialization."
	     "This is required because the sensors need to be powered before board initialization.");
```

nit: The assert message needs to reflect the error condition better. So consider rewording it like this.